encodings/alp/src/alp_rd/mod.rs

```diff
@@ -261,7 +261,7 @@ pub fn alp_rd_decode<T: ALPRDFloat>(
     left_parts: &[u16],
     left_parts_dict: &[u16],
     right_bit_width: u8,
-    right_parts: &[T::UINT],
+    right_parts: Vec<T::UINT>,
```

Comment:
We need a memory allocation of the same length and bit-width as the `right_parts`, so we take an owned copy. If the right_parts were just allocated (e.g. by decompressing a bitpacked array), then we save an allocation.